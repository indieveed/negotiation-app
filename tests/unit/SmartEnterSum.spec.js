import { shallowMount } from '@vue/test-utils'
import SmartEnterSum from '@/components/smart/enter-sum.vue'
import UINumberInput from '@/components/ui/number-input'


describe('ui/tabs.vue', () => {
  it('renders the component', () => {
    const wrapper = shallowMount(SmartEnterSum)
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('contains a number input', () => {
    const wrapper = shallowMount(SmartEnterSum)
    expect(wrapper.contains(UINumberInput)).toBe(true)
  })
  it('passes label prop to the input', () => {
    const wrapper = shallowMount(SmartEnterSum, {
      propsData: {
        label: 'foo'
      }
    })
    const input = wrapper.find(UINumberInput)
    expect(input.vm.label).toBe(wrapper.vm.label)
  })
})
