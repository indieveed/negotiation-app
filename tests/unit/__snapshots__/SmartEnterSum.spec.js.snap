// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/tabs.vue renders the component 1`] = `
<form class="enter-sum">
  <uinumberinput-stub placeholder="80000" label="Enter sum:" required="true"></uinumberinput-stub>
  <uibutton-stub class="enter-sum__submit">submit</uibutton-stub>
</form>
`;
